api/basic/get-pin-router.ts

```diff
@@ -0,0 +1,97 @@
+import { APIGatewayProxyEventV2 } from 'aws-lambda'
+import { ClusterStatusResponse, Response } from './schema.js'
+import { CID } from 'multiformats/cid'
+import fetch from 'node-fetch'
+
+import { doAuth } from './helper/auth-basic.js'
+
+/**
+ * AWS API Gateway handler for GET /pins/${cid}
+ * Collect the params and delegate to getPin to do the work
+ *
+ * We provide responses in Payload format v2.0
+ * see: https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html#http-api-develop-integrations-lambda.proxy-format
+ */
+export async function handler (event: APIGatewayProxyEventV2): Promise<Response> {
+  const {
+    CLUSTER_BASIC_AUTH_TOKEN: token = '',
+    INDEXER_ENDPOINT: indexerEndpoint = '',
+    PICKUP_ENDPOINT: pickupEndpoint = ''
+  } = process.env
+
+  const authResponse = doAuth(event.headers.authorization)
+  if (authResponse != null) return authResponse
+
+  const cid = event.pathParameters?.cid ?? ''
+
+  // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
+  if (!cid) {
```

Comment:
I'd do the validations at first and fail fast

Reply:
I've cleaned up the validation creating some helpers and given precedence to the readability.